beamer/health/check.py

```diff
@@ -0,0 +1,526 @@
+import logging
+from collections import defaultdict
+from dataclasses import dataclass, field
+from datetime import datetime
+from pathlib import Path
+from typing import DefaultDict, Optional, TypedDict, cast
+
+import toml
+from eth_typing import BlockNumber
+from typing_extensions import NotRequired
+from web3 import Web3
+from web3.constants import ADDRESS_ZERO
+from web3.middleware import geth_poa_middleware
+from web3.types import BlockData
+
+import beamer.agent.contracts
+import beamer.agent.events
+from beamer.agent.config import _merge_dicts
+from beamer.agent.events import ClaimMade, DepositWithdrawn, RequestCreated, RequestFilled
+from beamer.agent.typing import ChainId
+from beamer.agent.util import load_ERC20_abi, setup_logging
+from beamer.health.notification_state import NotificationState
+from beamer.health.notify import Message, NotificationConfig, Notify
+
+logging.disable()
+
+
+class MessageTypes:
+    REQUEST_EXPIRED = "RequestExpired"
+    DEPOSIT_WITHDRAWN = "DepositWithdrawn"
+    UNCLAIMED_FILL = "UnclaimedFill"
+    CHALLENGE_GAME = "ChallengeGame"
+    CHALLENGE_GAME_CLAIM_BY_SOMEONE_ELSE = "ChallengeGameClaimedBySomeoneElse"
+
+
+TokenMap = dict[str, list[tuple[str, str]]]
+
+
+class MonitorConfig(TypedDict):
```

Comment:
Should probably be renamed to `HealthConfig` or something like that now that we changed naming.